### PR TITLE
[Remove Action][Move Action] Update logic when working with locked, aliased domain objects

### LINKED ISSUE #6297: Shift Logs with one or more locked pages can't be moved
#### Summary
Shift Logs with one or more locked pages do not display the "Move" action, but should. Aliases to the same in layouts, Tab views, etc. also don't display that action, but _can_ be removed using those interfaces remove controls (toolbar buttons, tab "X" button).

#### Expected vs Current Behavior
The main problem to solve for here is shift logs included in compositions. Shift Logs, even with locked pages, should be able to be moved and removed via the appropriate actions in the Actions menu. The conundrum here is we should allow aliases to be moved/deleted at will, allow the "original" to be moved, but NOT allow it to be deleted.

#### Steps to Reproduce

1. Create a Shift Log, add an entry to a page within it.
2. Context-click the Shift Log object in the tree and note that "Move" and "Remove" are available actions in that items Actions menu.
3. Commit the page.
4. Perform step 2 again: observe that "Move" and "Remove" are not available in the Actions menu.

#### Environment

* Open MCT Version:  2.1.4
* Deployment Type:  MMOC
* OS: Linus
* Browser: Chrome

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #6297 

### Describe your changes:

- added objectAPI method `isAlias` since we use this pretty often
- updated move action to work on locked domainObjects
- updated remove action to allow removal of locked objects IF they are aliases
- updated existing unit tests

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by charlesh88 — CHANGES_REQUESTED** [review]
Using the Remove action from the layout frame of an object in a layout is problematic. 

Screen Shot 2023-03-06 at 12 39 36 PM

The resulting dialog is the one that is displayed when the user is operating on a domain object:

Screen Shot 2023-03-06 at 12 42 14 PM

However, this action is really more akin to the dialog that is displayed when using the layout's Trash toolbar button to remove the frame from the layout:
Screen Shot 2023-03-06 at 12 42 49 PM

Given that Shift Logs are critically important, I don't want there to be any ambiguity about what a remove result would be. Can we invoke a different dialog in this particular workflow case?

**Comment by jvigliotta** [comment]
@charlesh88 I'll take a look at the options for dialogs based on where the action is invoked. I'll reach out if I have questions. Thanks!

**Comment by jvigliotta** [comment]
@charlesh88 I've added special messaging for Display and Flexible 3-dot menu removal of items.

File: src/api/objects/ObjectAPI.js [hunk 1]
```diff
@@ -781,6 +781,28 @@ export default class ObjectAPI {
...
+     * Check if a child domainObject is an alias (link)
+     * by comparing the it's location to the parents identifier.
+     * If the location does not match the parent's identifier, then
+     * the domainObject is an alias.
+     *
+     * @param {DomainObject} childDomainObject
+     * @param {DomainObject} parentDomainObject
+     * @returns {boolean} true if alias, false if not
+     */
+    isAlias(childDomainObject, parentDomainObject) {
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:794]: Let's name this `isLink` since that is the nomenclature we've historically always used.

Comment (akhenry) [reply src/api/objects/ObjectAPI.js:794]: Thanks for making this an API function. I've no idea how we've made it this far without there being an API function for this 😄

File: src/plugins/move/MoveAction.js [hunk 2]
```diff
@@ -63,14 +63,14 @@ export default class MoveAction {
     async onSave(changes) {
         this.startTransaction();
 
-        let inNavigationPath = this.inNavigationPath(this.object);
+        const inNavigationPath = this.inNavigationPath(this.object);
+        const parentDomainObjectpath = changes.location || [this.parent];
+        const parent = parentDomainObjectpath[0];
+
         if (inNavigationPath && this.openmct.editor.isEditing()) {
             this.openmct.editor.save();
```
Comment (akhenry) [inline src/plugins/move/MoveAction.js:71]: Something about this logic doesn't seem right. We really leaving edit mode by triggering the move action?

File: src/plugins/move/MoveAction.js [hunk 3]
```diff
@@ -91,12 +91,13 @@ export default class MoveAction {
         }
 
         let newObjectPath;
+
         if (parentDomainObjectpath) {
             newObjectPath = parentDomainObjectpath && [this.object].concat(parentDomainObjectpath);
         } else {
+            const root = await this.openmct.objects.getRoot();
+            const rootChildCount = root.composition.length;
```
Comment (akhenry) [inline src/plugins/move/MoveAction.js:99]: Let's use the composition API please.

File: src/plugins/remove/RemoveAction.js [hunk 4]
```diff
@@ -39,28 +41,37 @@ export default class RemoveAction {
     }
 
     async invoke(objectPath) {
-        let object = objectPath[0];
-        let parent = objectPath[1];
+        const child = objectPath[0];
```
Comment (akhenry) [inline src/plugins/remove/RemoveAction.js:44]: THANK you. 🙏

**Review by akhenry — CHANGES_REQUESTED** [review]
Looking good! Really small suggestions. Thanks!

File: src/api/objects/ObjectAPI.js (hunk 1 above)
Comment (jvigliotta) [reply src/api/objects/ObjectAPI.js:794]: No problem! Yeah, `isLink` makes more sense!

File: src/plugins/move/MoveAction.js (hunk 2 above)
Comment (jvigliotta) [reply src/plugins/move/MoveAction.js:71]: Yeah that's been in there for a while, I'll test it out, if this is actually the case, which I think it is, is there another solution you'd like me to implement?

Comment (jvigliotta) [reply src/plugins/move/MoveAction.js:71]: Yep, just straight up saves whatever you had edited up to the point and moves it. I can add a confirmations dialog... I feel like that makes sense.

File: src/plugins/move/MoveAction.js (hunk 3 above)
Comment (jvigliotta) [reply src/plugins/move/MoveAction.js]: Gotcha!

File: src/plugins/move/MoveAction.js (hunk 2 above)
Comment (jvigliotta) [reply src/plugins/move/MoveAction.js]: Ok, removed for MoveAction, left it in RemoveAction as that's where it was originally introduced.

File: src/api/objects/ObjectAPI.js (hunk 1 above)
Comment (akhenry) [reply src/api/objects/ObjectAPI.js]: Oh wait, I just discovered `isObjectPathToLink` which is doing the same thing, but has a weird function signature. I don't know why it requires you to provide both an object _and_ an objectPath. Let's use the existing function and file a followup to refactor that function so it only requires an objectPath.

**Review by akhenry — CHANGES_REQUESTED** [review]
Just one small followup - https://github.com/nasa/openmct/pull/6384#discussion_r1137829952. Otherwise this is ready to go.

I discovered that we already have a method that does this, it's just a bit confusing. I'd rather not duplicate that functionality.

**Comment by akhenry** [comment]
Looks great, thanks!

**Review by akhenry — APPROVED** [review]

**Pull request merged**